pkg/podimpersonation/podimpersonation.go

```diff
@@ -559,9 +568,33 @@ func (s *PodImpersonation) augmentPod(pod *v1.Pod, sa *v1.ServiceAccount, secret
 				continue
 			}
 
+			vmount := v1.VolumeMount{
+				Name:      "user-kubeconfig",
+				MountPath: "/tmp/.kube",
+			}
+			cfgVMount := v1.VolumeMount{
+				Name:      "user-kubeconfig-map",
+				MountPath: "/home/.kube/config",
+				SubPath:   "config",
+			}
+
+			pod.Spec.InitContainers = []v1.Container{
+				{
+					Name:         "init-container",
+					Image:        "busybox",
+					Command:      []string{"sh", "-c", fmt.Sprintf("cp %s %s && chown %d %s/config", cfgVMount.MountPath, vmount.MountPath, shellUser, vmount.MountPath)},
+					Resources:    v1.ResourceRequirements{},
+					ResizePolicy: nil,
+					SecurityContext: &v1.SecurityContext{
+						RunAsUser:  &zero,
+						RunAsGroup: &zero,
+					},
+					VolumeMounts: []v1.VolumeMount{cfgVMount, vmount},
+				},
+			}
+
 			pod.Spec.Containers[i].VolumeMounts = append(container.VolumeMounts, v1.VolumeMount{
 				Name:      "user-kubeconfig",
-				ReadOnly:  true,
 				MountPath: envvar.Value,
 				SubPath:   "config",
```

Comment:
Do we need this subpath?

Reply:
We only use what's inside the `config` key of the ConfigMap. We don't _need_ it, but I think it's good to specify a `SubPath` if we know exactly what we need

Reply:
@rohitsakala Yes, sorry if I wasn't clear. I meant that we only copy the `config` from the `configMap` to this volume and that's why we only need to mount this path to the container